credentials/endpointcreds/provider.go

```diff
@@ -132,5 +163,23 @@ func (p *Provider) Retrieve(ctx context.Context) (aws.Credentials, error) {
 }
 
 func (p *Provider) getCredentials(ctx context.Context) (*client.GetCredentialsOutput, error) {
-	return p.client.GetCredentials(ctx, &client.GetCredentialsInput{AuthorizationToken: p.options.AuthorizationToken})
+	authToken := p.options.AuthorizationToken
+
+	var err error
+	if p.options.AuthorizationTokenProvider != nil {
+		authToken, err = p.options.AuthorizationTokenProvider.GetToken()
+		if err != nil {
+			return nil, fmt.Errorf("get authorization token: %v", err)
+		}
+	}
+
+	if strings.ContainsAny(authToken, "\r\n") {
+		return nil, fmt.Errorf("authorization token contains invalid newline sequence")
+	}
+
+	input := &client.GetCredentialsInput{}
+	if authToken != "" {
+		input.AuthorizationToken = authToken
+	}
+	return p.client.GetCredentials(ctx, input)
```

Comment:
You can condense this to like what it was before:

```suggestion
	return p.client.GetCredentials(ctx, &client.GetCredentialsInput{AuthorizationToken: authToken})
```

The zero value for `string` is already `""` - there's no need to check whether you ended up with something else in `authToken`.